src/platform_impl/linux/x11/event_processor.rs

```diff
@@ -429,6 +430,27 @@ impl EventProcessor {
             return;
         }
 
+        if xev.data.get_long(0) as xproto::Atom == wt.net_wm_sync_request {
+            let sync_counter_id = match self
+                .with_window(xev.window as xproto::Window, |window| window.sync_counter_id())
+            {
+                Some(sync_counter_id) => sync_counter_id,
+                None => return,
+            };
+
+            #[allow(clippy::identity_op)]
+            let lo = (bytemuck::cast::<c_long, c_ulong>(xev.data.get_long(2)) & 0xffffffff) as u32;
+            #[allow(clippy::identity_op)]
+            let hi = (bytemuck::cast::<c_long, c_ulong>(xev.data.get_long(3)) & 0xffffffff) as u32;
+            let hi = bytemuck::cast::<u32, i32>(hi);
```

Comment:
Do we lose anything by doing `as` instead of `bytemuck::cast`?

Reply:
The data's format is 32 bit numbers. On 64 bit Linux we could just not skip the `c_ulong` conversion entirely, but on 32 bit Linux the `c_long` and `c_ulong` types are 32 bit, so the last bit being the sign becomes a problem. So this is done to make sure both 64 bit and 32 bit interpret the data correctly.

Alternatively I could make 2 separate codepaths, one for 32 bit and another for 64 bit. Might be easier to read?